test/e2e/suites/base/clustertaintpolicy_test.go

```diff
@@ -0,0 +1,356 @@
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package base
+
+import (
+	"fmt"
+	"time"
+
+	"github.com/onsi/ginkgo/v2"
+	"github.com/onsi/gomega"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/rand"
+
+	clusterv1alpha1 "github.com/karmada-io/karmada/pkg/apis/cluster/v1alpha1"
+	policyv1alpha1 "github.com/karmada-io/karmada/pkg/apis/policy/v1alpha1"
+	"github.com/karmada-io/karmada/pkg/util/helper"
+	"github.com/karmada-io/karmada/test/e2e/framework"
+	karmadaresource "github.com/karmada-io/karmada/test/e2e/framework/resource/karmada"
+)
+
+var _ = framework.SerialDescribe("Taint cluster with ClusterTaintPolicy", func() {
+	var policyName string
+	var clusterTaintPolicy *policyv1alpha1.ClusterTaintPolicy
+
+	ginkgo.JustBeforeEach(func() {
+		karmadaresource.CreateClusterTaintPolicy(karmadaClient, clusterTaintPolicy)
+		ginkgo.DeferCleanup(func() {
+			karmadaresource.RemoveClusterTaintPolicy(karmadaClient, policyName)
+		})
+	})
+
+	ginkgo.Context("Disable cluster object by update the APIEndpoint", func() {
+		ginkgo.BeforeEach(func() {
+			policyName = clusterTaintPolicyNamePrefix + rand.String(RandomStrLength)
+			clusterTaintPolicy = &policyv1alpha1.ClusterTaintPolicy{
+				TypeMeta: metav1.TypeMeta{
+					APIVersion: policyv1alpha1.SchemeGroupVersion.String(),
+					Kind:       "ClusterTaintPolicy",
+				},
+				ObjectMeta: metav1.ObjectMeta{
+					Name: policyName,
+				},
+				Spec: policyv1alpha1.ClusterTaintPolicySpec{
+					TargetClusters: &policyv1alpha1.ClusterAffinity{
+						LabelSelector: &metav1.LabelSelector{
+							// Only test push mode cluster, because pull mode
+							// clusters cannot be disabled by changing APIEndpoint.
+							MatchLabels: pushModeClusterLabels,
+						},
+					},
+					AddOnConditions: []policyv1alpha1.MatchCondition{
+						{
+							ConditionType: clusterv1alpha1.ClusterConditionReady,
+							Operator:      policyv1alpha1.MatchConditionOpNotIn,
+							StatusValues:  []metav1.ConditionStatus{metav1.ConditionTrue},
+						},
+					},
+					RemoveOnConditions: []policyv1alpha1.MatchCondition{
+						{
+							ConditionType: clusterv1alpha1.ClusterConditionReady,
+							Operator:      policyv1alpha1.MatchConditionOpIn,
+							StatusValues:  []metav1.ConditionStatus{metav1.ConditionTrue},
+						},
+					},
+					Taints: []policyv1alpha1.Taint{
+						{
+							Key:    "testing/not-ready",
+							Effect: "NoSchedule",
+						},
+					},
+				},
+			}
+		})
+
+		ginkgo.It("Trigger tainting cluster by making cluster not ready", func() {
```

Comment:
Do we alreay have the test check the condition in case the cluster becomes unreachable?
This case is testing another feature, for the cluster-taint-policy-controller it don't care how the conditions be added.